File: experiments/source/cpp/relu_tupled.cpp
#include <loom/loom.hpp>

LOOM_ENTRY loom::Tensor1 relu_tupled(loom::Tensor1 x) {
  auto thresholds = loom::tuple(0.0f, 1.0f);
  auto [zero, _one] = thresholds;
  auto clip = [zero](float xi) -> float { return xi > zero ? xi : zero; };
  return loom::Tensor1::map(clip, x);
}

File: experiments/source/python/relu_tupled.py
import loom


@loom.entry
def relu_tupled(x: loom.Tensor1) -> loom.Tensor1:
    thresholds = (0.0, 1.0)
    zero, _one = thresholds

    def clip(xi: float) -> float:
        return xi if xi > zero else zero

    return loom.Tensor1.map(clip, x)
